core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncer.java

```diff
@@ -108,9 +107,7 @@ public void heartbeat() throws InterruptedException {
                 }, RetryUtils.defaultActiveSyncClientRetry());
               }
               // Journal the latest processed txId
-              if (syncResult) {
-                mFileSystemMaster.recordActiveSyncTxid(syncInfo.getTxId(), mMountId);
-              }
+              mFileSystemMaster.recordActiveSyncTxid(syncInfo.getTxId(), mMountId);
```

Comment:
how do we prevent the race condition where the mount point is deleted before this is written? When we replay, this would be an invalid mount id

Reply:
so I took another closer look at the code. before we unmount, we actually stop this polling thread, so there can not be such a race condition.